packages/headless/src/features/configuration/configuration-slice.ts

```diff
@@ -176,18 +90,170 @@ function handleUpdateBasicConfiguration(
   state: ConfigurationState,
   payload: UpdateBasicConfigurationActionCreatorPayload
 ) {
-  if (payload.accessToken) {
+  if (!isNullOrUndefined(payload.accessToken)) {
     state.accessToken = payload.accessToken;
   }
-  if (payload.organizationId) {
+
+  state.environment = payload.environment ?? 'prod';
+
+  if (!isNullOrUndefined(payload.organizationId)) {
     state.organizationId = payload.organizationId;
   }
-  if (payload.platformUrl) {
-    state.platformUrl = payload.platformUrl;
-    state.search.apiBaseUrl = `${payload.platformUrl}${searchAPIEndpoint}`;
-    state.analytics.apiBaseUrl = legacyAnalyticsUrlFromPlatformUrl(
-      payload.platformUrl,
-      state.organizationId
+
+  if (
```

Comment:
The idea here is that if the `updateBasicConfiguration` action payload contains an `organizationId` and/or an `environment, then it means that the `platformUrl` _will_ change, and that the Search API, Analytics API (next), and Commerce API base URLs _may_ change.

The `platformUrl` is always going to be `https://<orgId>.org<environment|>.coveo.com`

The various `apiBaseUrl` values will typically be updated accordingly. However, if we detect that one or more of them don't match the expected format of a default Coveo org endpoint, then we assume that this endpoint has been overridden through a proxyBaseUrl and we don't reset it to the default.

Reply:
When will this happen? When would we call `handleUpdateBasicConfiguration` AFTER we've already set a proxyBaseUrl?

And why isn't the basicConfiguration staying "basic" and we should be looking at the advanced options like `proxyBaseUrl` more at the time of making the requests?

AKA, if I set a proxy, and then update my orgId/environment in the state, it shouldn't matter and should be superceded by having set a `proxyBaseUrl`. I don't think it's the right place to handle this situation, we should update the Search API, Analytics API (next), and Commerce API **base** URLs, keyword here being base.

But the respective API clients should be looking for a proxyBaseUrl before looking at their baseUrls.

Reply:
> When will this happen? When would we call handleUpdateBasicConfiguration AFTER we've already set a proxyBaseUrl?

If you manually dispatch the action. Granted, this is an edge case.

> And why isn't the basicConfiguration staying "basic" and we should be looking at the advanced options like proxyBaseUrl more at the time of making the requests?

That's a good idea! I'll rework this.